include/dxc/Support/HLSLOptions.td

```diff
@@ -168,6 +168,10 @@ def opt_enable : Separate<["-", "/"], "opt-enable">, Group<hlsloptz_Group>, Flag
 def opt_select : MultiArg<["-", "/"], "opt-select", 2>, MetaVarName<"<opt> <variant>">, Group<hlsloptz_Group>, Flags<[CoreOption, DriverOption, HelpHidden]>,
   HelpText<"Select this optimization variant.">;
 
+def fdisable_debug_info : Flag<["-"], "fdisable-debug-info">,
+  Group<hlslcomp_Group>, Flags<[CoreOption]>,
+  HelpText<"Force disabling all debug info. This will break diagnostic generation for late validation.">;
```

Comment:
I'm not sure the description needs to mention debug info. It might be more confusing since the flag to control the debug info in the output is `-Zi`. In this case, it seems like an implementation detail. Of more interest might be to refer to it as not maintaining line tracking information and mention it improves compile times. The diagnostic helper functions should still give the best information about location they can when they don't have line info.